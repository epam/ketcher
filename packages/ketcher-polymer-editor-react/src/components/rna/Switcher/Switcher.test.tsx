/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { fireEvent, screen, render } from '@testing-library/react';

import { Switcher } from './';

describe('RNA Switcher component', () => {
  const mockSetActiveMonomerType = jest.fn();
  const buttons = ['R(A)P', 'R', 'A', 'P'];

  it('should call click event handler on each button when clicked', () => {
    render(
      withThemeProvider(
        <Switcher
          selectedMonomers={['R', 'A', 'P']}
          setActiveMonomerType={mockSetActiveMonomerType}
        />,
      ),
    );
    buttons.forEach((button, index) => {
      const buttonR = screen.getByText(button);
      fireEvent.click(buttonR);
      expect(mockSetActiveMonomerType.mock.calls.length).toEqual(index + 1);
    });
  });

  it('should render correctly with each button selected', () => {
    const view = render(
      withThemeProvider(
        <Switcher
          selectedMonomers={['R', 'A', 'P']}
          setActiveMonomerType={mockSetActiveMonomerType}
        />,
      ),
    );
    buttons.forEach((button) => {
      const buttonR = screen.getByText(button);
      fireEvent.click(buttonR);
      expect(view).toMatchSnapshot();
    });
  });

  it('should reset to initial state when reset button clicked', () => {
    render(
      withThemeProvider(
        <Switcher
          selectedMonomers={['R', 'A', 'P']}
          setActiveMonomerType={mockSetActiveMonomerType}
        />,
      ),
    );
    const sugarBtn = screen.getByText(buttons[1]);
    fireEvent.click(sugarBtn);
    const resetBtn = screen.getByText('Reset');
    fireEvent.click(resetBtn);
    const nucleotideBtn = screen.getByText(buttons[0]);
    expect(nucleotideBtn).toHaveStyle(`background-color: rgb(22, 119, 130)`);
  });
});
