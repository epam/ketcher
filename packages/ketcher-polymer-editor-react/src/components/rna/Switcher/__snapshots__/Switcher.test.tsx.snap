// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RNA Switcher component should render correctly with each button selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1m326po"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R(A)P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="css-11l5obs"
        >
          <icon-rap-left-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
          <icon-rap-middle-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
          <icon-rap-right-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
        </div>
        <div
          class="css-11ydt2g"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            R
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            A
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            P
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reset
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1m326po"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
        tabindex="0"
        type="button"
      >
        R(A)P
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="css-11l5obs"
      >
        <icon-rap-left-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
        <icon-rap-middle-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
        <icon-rap-right-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
      </div>
      <div
        class="css-11ydt2g"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          A
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RNA Switcher component should render correctly with each button selected 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1m326po"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R(A)P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="css-11l5obs"
        >
          <icon-rap-left-link.svg
            classname="css-1eyukqs"
            fill="#333333"
            role="img"
          />
          <icon-rap-middle-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
          <icon-rap-right-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
        </div>
        <div
          class="css-11ydt2g"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            R
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            A
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            P
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reset
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1m326po"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
        tabindex="0"
        type="button"
      >
        R(A)P
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="css-11l5obs"
      >
        <icon-rap-left-link.svg
          classname="css-1eyukqs"
          fill="#333333"
          role="img"
        />
        <icon-rap-middle-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
        <icon-rap-right-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
      </div>
      <div
        class="css-11ydt2g"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          A
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RNA Switcher component should render correctly with each button selected 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1m326po"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R(A)P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="css-11l5obs"
        >
          <icon-rap-left-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
          <icon-rap-middle-link.svg
            classname="css-1eyukqs"
            fill="#333333"
            role="img"
          />
          <icon-rap-right-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
        </div>
        <div
          class="css-11ydt2g"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            R
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            A
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            P
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reset
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1m326po"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
        tabindex="0"
        type="button"
      >
        R(A)P
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="css-11l5obs"
      >
        <icon-rap-left-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
        <icon-rap-middle-link.svg
          classname="css-1eyukqs"
          fill="#333333"
          role="img"
        />
        <icon-rap-right-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
      </div>
      <div
        class="css-11ydt2g"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          A
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`RNA Switcher component should render correctly with each button selected 4`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1m326po"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R(A)P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="css-11l5obs"
        >
          <icon-rap-left-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
          <icon-rap-middle-link.svg
            classname="css-1vr3czz"
            fill="#333333"
            role="img"
          />
          <icon-rap-right-link.svg
            classname="css-1eyukqs"
            fill="#333333"
            role="img"
          />
        </div>
        <div
          class="css-11ydt2g"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            R
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            A
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
            tabindex="0"
            type="button"
          >
            P
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reset
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1m326po"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
        tabindex="0"
        type="button"
      >
        R(A)P
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="css-11l5obs"
      >
        <icon-rap-left-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
        <icon-rap-middle-link.svg
          classname="css-1vr3czz"
          fill="#333333"
          role="img"
        />
        <icon-rap-right-link.svg
          classname="css-1eyukqs"
          fill="#333333"
          role="img"
        />
      </div>
      <div
        class="css-11ydt2g"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          R
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-71ofci-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          A
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1ii4wrn-MuiButtonBase-root-MuiButton-root-RAPButton"
          tabindex="0"
          type="button"
        >
          P
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1l3rijb-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
